toolchain/sem_ir/stringify_type.cpp

```diff
@@ -321,29 +321,40 @@ auto StringifyTypeExpr(const SemIR::File& sem_ir, InstId outer_inst_id)
         break;
       }
       case CARBON_KIND(FunctionType inst): {
-        auto fn_name_id = sem_ir.functions().Get(inst.function_id).name_id;
-        out << "<type of " << sem_ir.names().GetFormatted(fn_name_id) << ">";
+        const auto& fn = sem_ir.functions().Get(inst.function_id);
+        out << "<type of ";
+        step_stack.PushString(">");
+        step_stack.PushQualifiedName(fn.parent_scope_id, fn.name_id);
         break;
       }
-      case CARBON_KIND(ImplFunctionType inst): {
-        auto fn_inst =
-            sem_ir.insts().GetAs<FunctionType>(inst.interface_function_type_id);
-        auto fn_name_id = sem_ir.functions().Get(fn_inst.function_id).name_id;
-        out << "<type of " << sem_ir.names().GetFormatted(fn_name_id) << " in ";
+      case CARBON_KIND(FunctionTypeWithSelfType inst): {
```

Comment:
Sorry, I missed this on my first pass. Is this tested somewhere?

It looks like this will format as `<type of in self_id>interface_function_type_id` -- shouldn't `fn_inst` be examined last?. Looking through tests (`git grep "type of" **/*.carbon`), I see `<type of F>` in `check/testdata/operators/builtin/fail_redundant_compound_access.carbon` as the only `<type of` that's tested. Not sure how easy this is to test, but if you have ideas it might be good to add something (also for the surrounding changes, since I think only `FunctionType` is tested, and nothing qualified).

Reply:
Thanks, fixed and tested in `toolchain/check/testdata/interface/no_prelude/fail_assoc_fn_invalid_use.carbon`.

Reply:
Minor and maybe not specific to this PR, but `<type of F in T as I>` might be clearer as `<type of F in (T as I)>` to avoid confusion about `<type of (F in T) as I>`. But I don't know if the `as` is always there, so maybe not a clear fix... I mean, `<type of F in (X)>` would be not-so-good, and the current format is probably usually clear enough.